distributed/protocol.py

```diff
@@ -64,35 +63,89 @@
     default_compression = 'lz4'
 
 
+def extract_big_bytes(x):
+    big = {}
+    _extract_big_bytes(x, big)
+    if big:
+        x = deepcopy(x)
+        for path in big:
+            t = get_in(path[:-1], x)
+            if isinstance(t, dict):
+                del t[path[-1]]
+            else:
+                t[path[-1]] = None
+    return x, big
+
+
+def _extract_big_bytes(x, big, path=()):
+    if type(x) is dict:
+        for k, v in x.items():
+            if isinstance(v, (list, dict)):
+                _extract_big_bytes(v, big, path + (k,))
+            elif type(v) is bytes and len(v) >= 2**31:
```

Comment:
We want to extract bytestrings that are even moderately large.  Probably anything larger than 1MB.  It would be nice to add tests to ensure that this was the case.

In particular, the following test might be useful:

```python
data = np.random.randint(0, 255, dtype='u1', size=(2000000)).data.tobytes()
msg = {'x': [{'y': data}]}
frames = dumps(msg)
assert any(data == frame for frame in frames)
```